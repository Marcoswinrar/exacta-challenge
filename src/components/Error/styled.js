import styled from 'styled-components'

export const ErrorMessage = styled.span`
  margin-bottom: 6px;
  color: crimson;

  :before {
    content: '*';
  }

`