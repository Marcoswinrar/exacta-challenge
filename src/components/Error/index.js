import { ErrorMessage } from "./styled"

const Error = ({ message }) => (
  <ErrorMessage className="error">{message}</ErrorMessage>
)

export default Error