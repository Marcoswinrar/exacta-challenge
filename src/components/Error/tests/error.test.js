import { render, screen } from "@testing-library/react"
import Error from ".."

describe("Componente <Error />", () => {

  it("Deve renderizar o component de erro sem problemas", () => {
    const {container} = render(<Error />)

    expect(container).toMatchSnapshot()
  })

  it("Deve mostrar a mensagem de erro passado ao componente", () => {
    render(<Error message={"Erro ao cadastrar o usuário"}/>)

    expect(screen.queryByText(/Erro ao cadastrar o usuário/i)).toBeInTheDocument()
  })
})