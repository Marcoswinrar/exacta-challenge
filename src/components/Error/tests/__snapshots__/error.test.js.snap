// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente <Error /> Deve renderizar o component de erro sem problemas 1`] = `
<div>
  <span
    class="sc-bczRLJ dNocsd error"
  />
</div>
`;
